components/ai_chat/resources/page/state/data-context-provider.tsx

```diff
@@ -336,6 +336,8 @@ function DataContextProvider(props: DataContextProviderProps) {
   }
 
   const isMobile = React.useMemo(() => loadTimeData.getBoolean('isMobile'), [])
+  const hasInitialHistory = React.useMemo(() =>
```

Comment:
I know you're just following what this file does but I don't think there's any point to this being in a `useMemo` as:
1. It isn't an expensive calculation (so we could just store it in a normal variable)
2. It will never change

Personally, I'd probably store the variable outside the component as 

```ts
const hasInitialHistory = loadTimeData.getBoolean('hasInitialHistory')
function DataContextProvider(...) {
    ...
}
```

Reply:
Really this shouldn't be part of the context at all as its completely static data, but I guess its fine as part of the interface the `input_box` expects.